docs/2.9.1/docs/app-dev/explicit-contract-disclosure.rst

```diff
@@ -264,3 +264,50 @@ These last two steps are executed using the new Daml Script functions supporting
 
 .. note:: For an example using Java bindings for client applications, see the
   `Java Bindings StockExchange example project <https://github.com/digital-asset/ex-java-bindings/blob/f474ae83976b0ad197e2fabfce9842fb9b3de907/StockExchange/README.rst>`_.
+
+Safeguards
+----------
+
+In the example above, what if the **Buyer** is malicious and wants to pay less than the official price quotation for **Seller**'s stock?
+**Buyer** might try to do so by modifying the received ``disclosedPriceQuotation`` payload received from the **StockExchange** by setting a lower value in the contract's arguments
+and then using the forged payload as a disclosed contract in the command submission exercising ``Offer_Accept`` on **Seller**'s offer.
+
+Contract authentication
+```````````````````````
+
+Scenarios like the one exemplified above are not possible due to a new technical feature introduced with explicit contract disclosure: Daml contract authentication.
+
+More specifically, each contract's information, such as its arguments, template-id, signatories, etc. are authenticated by
+incorporating in the contract's contract-id a hash over the relevant contract information, ensuring
```

Comment:
```suggestion
More specifically, each contract's arguments, template-id, signatories, keys, etc. are incorporated into 
the contract's contract-id. It is a hash over all the relevant information, ensuring
```

Reply:
`It is a hash over` in this edit should probably be replaced with `The contract ID hashes over` - I wasn't sure what `it` referred to in my first reading.